bumble/device.py

```diff
@@ -1717,6 +1731,9 @@ def on_connection_failure(self, connection_handle, error_code):
             'hci',
             HCI_Constant.error_name(error_code)
         )
+        error.transport = transport
```

Comment:
I suggest making `transport` a constructor parameter for `ConnectionError`

Reply:
I was doing the change but ConnectionError is also used here https://github.com/google/bumble/blob/main/bumble/rfcomm.py#L637
Do I still proceed ?

Reply:
I did the change, lmk what you think :+1: